flake.nix

```diff
@@ -16,7 +16,133 @@
         "x86_64-darwin" = [ "darwin" "x64" ];
         "aarch64-darwin" = [ "darwin" "arm64" ];
       };
+
     in
+    {
+      nixosModules.default = { config, ... }: with nixpkgs; with lib;
+      {
+        options = {
+          services.polykey = {
+            enable = mkEnableOption "Enable the Polykey agent. Users with the `polykey` group or root permissions will be able to manage the agent.";
+
+            recovery = mkEnableOption "Enable recovery mode for the node. Will try to source from a file under the configuration directory called `recovery-code`.";
+
+            configPath = mkOption {
+              type = with types; uniq str;
+              default = "/etc/polykey";
+              description = ''
+              The path to Polykey configuration directory.
+              A file called `password` with your desired password must be placed inside of this directory to set the node password.
+              '';
+            };
```

Comment:
How are we handling permissions for this password file? It's possible that the user makes this file to permissive which would be problematic. SSH handles this by refusing to use a file depending on it's mask. Can we do the same here?

Reply:
Currently, its not handling permissions in an ideal manner, it will just attempt to read the password file regardless of permissions. I believe its possible to cause the module to throw an error if a condition is met, so it should be possible to check the mask and throw an error if it isn't suitable. I'll get this done ASAP.

Reply:
I think we could add this to the CLI itself? Doing this through scripting seems very hacky.

Reply:
@CMCDragonkai do you think we should make it so PKCLI will refuse to accept any password files as an argument if they don't contain a proper file permission mask?

Reply:
If we have it at all it should be a check inside of `PK-CLI`

Reply:
I do not like the idea of a PK config directory. It should just be direct options in Nix or references to arbitrary file paths the user has set. We do not use config directories or config files as a rule for PK.